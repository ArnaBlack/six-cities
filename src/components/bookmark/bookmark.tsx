import * as React from 'react';
import withFavoriteOperation from '../../hocs/with-favorite-operation/with-favorite-operation';

interface Props {
  id: number,
  isFavorite: boolean,
  className: string,
  width: number,
  height: number,
  onClick: () => void,
}

const Bookmark = (props: Props) => {
  const {
    className,
    isFavorite,
    width,
    height,
    onClick,
  } = props;
  const activeClass = isFavorite ? `${className}__bookmark-button--active` : ``;

  return <button
    className={`${className}__bookmark-button button ${activeClass}`}
    type="button"
    onClick={onClick}
  >
    <svg
      className={`${className}__bookmark-icon`}
      width={width}
      height={height}
    >
      <use xlinkHref="#icon-bookmark"/>
    </svg>
    <span className="visually-hidden">To bookmarks</span>
  </button>;
};

export {Bookmark};
export default withFavoriteOperation(Bookmark);
