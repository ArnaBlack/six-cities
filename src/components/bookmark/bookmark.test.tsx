import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Bookmark} from './bookmark';

const mock = {
  id: 1,
  isFavorite: false,
  className: ``,
  width: 18,
  height: 19,
};

it(`Bookmark renders correctly`, () => {
  const {
    id,
    isFavorite,
    className,
    width,
    height,
  } = mock;
  const onClick = jest.fn();
  const updateFavorites = jest.fn();
  const tree = renderer
    .create(<Bookmark
      id={id}
      isFavorite={isFavorite}
      className={className}
      width={width}
      height={height}
      onClick={onClick}
      updateFavorites={updateFavorites}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
