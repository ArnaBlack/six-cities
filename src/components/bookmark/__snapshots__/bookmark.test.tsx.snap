// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark renders correctly 1`] = `
<button
  className="__bookmark-button button "
  onClick={[Function]}
  type="button"
>
  <svg
    className="__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
